modular_exponential/modular_exponential.py

```diff
@@ -0,0 +1,16 @@
+def mod_expo(_base, _pow, _mod):
+    res = 1  # Initialize result
+    _base = _base % _mod  # Update base if it is more than or equal _mod
+    while _pow > 0:
+        if _pow & 1:  # if _pow is odd multiply it with result
+            res = (res * _base) % _mod
+        _pow = _pow >> 1  # _pow must be even now
```

Comment:
They won't shadow anything until you have imported `*` from `math`.

Also in Python, variables starting with an underscore are considered as weak internal use indicator. So, `base_`, `pow_`, etc. should be used instead.